arcgis-ios-sdk-samples/Maps/Display device location with NMEA data sources/DisplayDeviceLocationWithNMEADataSourcesViewController.swift

```diff
@@ -24,31 +24,129 @@ class DisplayDeviceLocationWithNMEADataSourcesViewController: UIViewController {
+                // Skip any device which protocol is not included in the plist.
+                // This typically shouldn't happen, unless the device requires
+                // additional configuration.
+                continue
+            }
+            // Only return the first connected and supported accessory.
+            return (accessory, protocolString)
+        }
+        return nil
+    }
+    
+    /// The Bluetooth accessory picker connected to a supported accessory.
+    func accessoryDidConnect(connectedAccessory: EAAccessory, protocolString: String) {
+        nmeaLocationDataSource = AGSNMEALocationDataSource(eaAccessory: connectedAccessory, protocol: protocolString)
+        nmeaLocationDataSource?.locationChangeHandlerDelegate = self
+        start()
+    }
+    
+    @IBAction func chooseDataSource(_ sender: UIBarButtonItem) {
+        let alertController = UIAlertController(
+            title: "Choose an NMEA data source.",
+            message: nil,
+            preferredStyle: .actionSheet
+        )
+        // Add real data source to the options.
+        let realDataSourceAction = UIAlertAction(title: "Device", style: .default) { [self] _ in
+            if let (accessory, protocolString) = firstSupportedAccessoryWithProtocol() {
+                // Use the supported accessory directly if it's already connected.
+                accessoryDidConnect(connectedAccessory: accessory, protocolString: protocolString)
+            } else {
+                // Display an alert to pair the device with a Bluetooth accessory.
+                EAAccessoryManager.shared().showBluetoothAccessoryPicker(withNameFilter: nil) { error in
+                    if let error = error as? EABluetoothAccessoryPickerError {
+                        switch error.code {
+                        case .alreadyConnected:
+                            presentAlert(message: "The specified accessory was already connected.")
+                        case .resultNotFound:
+                            presentAlert(message: "The specified accessory could not be found, perhaps because it was turned off prior to connection.")
+                        case .resultCancelled:
+                            // Don't show error message if user cancels the picker.
+                            return
+                        default:
+                            presentAlert(message: "Selecting an accessory failed for an unknown reason.")
+                        }
+                    } else if let (accessory, protocolString) = firstSupportedAccessoryWithProtocol() {
+                        // Proceed with supported and connected accessory, and
+                        // ignore other accessories that aren't supported.
+                        accessoryDidConnect(connectedAccessory: accessory, protocolString: protocolString)
+                    }
+                }
+            }
+        }
```

Comment:
A few things to notice (and feedback needed)

- The 2 helper method names: `firstSupportedAccessoryWithProtocol` and `accessoryDidConnect` might not be the best naming pattern, but are the best that I can came up with 😅 
- Since there is no explicit disconnect operation from the iOS side, we have to check if the phone is already connected to some BT device. There are several states:
    - Not connected. Then we can prompt the BT picker
    - Connected, but to some other BT device such as headphone. Will need to present picker here.
    - Connected to a GNSS surveyor. Then we can directly instantiate the NMEA LDS from that accessory.
- Do we need the error code handling? I found the generic error message not helpful, but there aren't error description associated with these codes, so I copy and pasted from [here](https://developer.apple.com/documentation/externalaccessory/eabluetoothaccessorypickererror/code)
- Please revisit the previous comment on hardcoding protocol strings: https://github.com/Esri/arcgis-runtime-samples-ios/pull/1058#discussion_r616010708